test/lint/lint-python.py

```diff
@@ -30,28 +30,8 @@
 
 ENABLED = (
     'E101,'  # indentation contains mixed spaces and tabs
-    'E112,'  # expected an indented block
-    'E113,'  # unexpected indentation
-    'E115,'  # expected an indented block (comment)
-    'E116,'  # unexpected indentation (comment)
-    'E125,'  # continuation line with same indent as next logical line
-    'E129,'  # visually indented line with same indent as next logical line
-    'E131,'  # continuation line unaligned for hanging indent
-    'E133,'  # closing bracket is missing indentation
-    'E223,'  # tab before operator
-    'E224,'  # tab after operator
-    'E242,'  # tab after ','
-    'E266,'  # too many leading '#' for block comment
-    'E271,'  # multiple spaces after keyword
-    'E272,'  # multiple spaces before keyword
-    'E273,'  # tab after keyword
-    'E274,'  # tab before keyword
-    'E275,'  # missing whitespace after keyword
-    'E304,'  # blank lines found after function decorator
-    'E306,'  # expected 1 blank line before a nested definition
```

Comment:
In commit: [lint: Remove python whitespace and shadowing lint rules](https://github.com/bitcoin/bitcoin/pull/30723/commits/fafb0f62175798c5d5c174213c1ff1ccf5cea6ca)

+1 on dropping checks that result in `IndentError`

some of the other checks removed here seem reasonable to me e.g. E125, and E275, but not available/hidden behind preview flag in ruff so I'm ~0 on whether or not to remove the rest of them